File: test/unit/specs/ItinGen.spec.js
import Vue from 'vue'
import ItinGen from '@/components/ItinGen'

describe('ItinGen.vue', () => {
  const Constructor = Vue.extend(ItinGen)
  const vm = new Constructor().$mount()

  it('should render the correct header', () => {
    expect(vm.$el.querySelector('.header h1').textContent)
      .toEqual('Itinerary Generator')
  })

  it('should render the map', () => {
    expect(vm.$el.querySelector('.app div'))
      .contains('.vue-map-container')
  })

  it('should display a location', () => {
    expect(vm.$el.querySelector('.app div'))
      .contains('.vue-map-marker')
  })

  it('should display a sidebar', () => {
    expect(vm.$el.querySelector('.app div'))
      .contains('.sidebar')
  })

  it('should display a login button', () => {
    expect(vm.$el.querySelector('.sidebar button').textContent)
      .toEqual('Login')
  })

  it('should display a sign up button', () => {
    expect(vm.$el.querySelector('sidebar button').textContent)
      .toEqual('Signup')
  })

  it('should display an itinerary', () => {
    expect(vm.$el.querySelector('sidebar div'))
      .containes('.itinerary')
  })

  it('should render the generate button', () => {
    expect(vm.$el.querySelector('.generate button').textContent)
      .toEqual('Generate')
  })

  it('should render the dislike button', () => {
    expect(vm.$el.querySelector('.dislike button').textContent)
      .toEqual('Dislike')
  })

  it('should render a save current itinerary button', () => {
    expect(vm.$el.querySelector('.save button').textContent)
      .toEqual('Save Current Itinerary')
  })

  it('should render a link to saved itineraries', () => {
    expect(vm.$el.querySelector('.saved button').textContent)
      .toEqual('View Saved Itineraries')
  })

})

File: spec/features/itingen_spec.rb
require 'rails_helper'

describe 'content' do
  before do
    visit root_path
  end

  describe 'homepage' do
    it 'should be reached successfully' do
      expect(page.status_code).to eq(200)
    end

    describe 'maps' do
      it 'should display google maps' do
        page.should have_css('div', :id => 'map')
      end

      it 'should display a location' do
        page.should have_css('h5', :class => 'card-title')
      end
    end

    describe 'sidebar' do
      describe 'menu' do
        it 'should have a menu' do
          page.should have_css('div', :id => 'mySidebar')
        end

        it 'should have an option to login' do
          page.should have_css('button', :text => 'Login')
        end

        it 'should have an option to sign up' do
          page.should have_css('button', :text => 'Register')
        end
      end

      it 'should have an itinerary' do
        page.should have_css('div', :class => 'card-header')
      end

      it 'should have a like button' do
        page.should have_css('a', :id => 'like')
      end

      it 'should have a dislike button' do
        page.should have_css('a', :id => 'dislike')
      end

      it 'should have a save current itinerary button' do
        page.should have_css('a', :id => 'like')
      end

      it 'should have a link to saved itineraries' do
        page.should have_css('button', :text => 'Liked Itineraries')
      end

    end
  end
end
